crates/librqbit/src/torrent_state/live/mod.rs

```diff
@@ -1657,4 +1671,21 @@ impl PeerHandler {
             .context("error sending UtMetadata: channel closed")?;
         Ok(())
     }
+
+    fn on_pex_message<B>(&self, msg: UtPex<B>)
+    where
+        B: AsRef<[u8]> + std::fmt::Debug,
+    {
+        // TODO: this is just first attempt at pex - will need more sophistication on adding peers - BEP 40,  check number of live, seen peers ...
+        if let Ok(peers) = msg.added_peers() {
+            peers.for_each(|peer| {
```

Comment:
Clearly adding eagerly is not good  - it takes huge amount of CPU - will need to find something better

Reply:
What eats CPU exactly out of this?

Reply:
Adding eagerly should be the way to go, just need to fix our workaround the issues